test/rpm/utils.sh

```diff
@@ -4,136 +4,203 @@ set -euo pipefail
 
 source "$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)/../ci/utils.sh"
 
-rpm_config_base_dir="/etc/go-fdo-server"
-rpm_sysconfig_dir="/etc/sysconfig"
-rpm_group="go-fdo-server"
-
-rpm_device_ca_user="go-fdo-server-manufacturer"
-rpm_device_ca_crt="${rpm_config_base_dir}/device-ca.crt"
-rpm_device_ca_key="${rpm_config_base_dir}/device-ca.key"
+# PLEASE READ:
+#
+# The FMF tests deploy the FDO servers via RPM packages. These
+# packages are either provided by the Packit service or pulled from
+# the fedora-iot COPR repository.
+#
+# All test-related configuration resides under the ${base_dir} working
+# directory, which is made available as a testing artifact for
+# debugging purposes. The FDO server configuration files also reside
+# in ${base_dir} and are copied into the server configuration search
+# path at the start of the test.
+#
+# If a test requires a custom configuration consider the following
+# recommendations:
+#
+# o) Keep all test configuration in ${base_dir} in order to get it
+# included in the testing artifacts.
+#
+# o) To create a custom configuration file copy the necessary
+# `generate_${service}_config()` function(s) to your test and modify
+# them to produce the desired test configuration. Your generated
+# configuration file will automatically be saved in a directory that
+# takes precedence over the default in the server's configuration
+# search path. A copy of configuration file will be available
+# as a testing artifact end of the test.
+#
+# o) To pass command line arguments to the server create a systemd
+# "drop-in" file that overrides the ExecStart= setting in the service
+# file. This drop-in file should be placed in the directory defined by
+# the variable `systemd_${service}_drop_in_dir`.  Remember to run
+# `sudo systemctl daemon-reload` after writing the drop-in file.
+
+configs_dir="${base_dir}/configs"
+directories+=("${configs_dir}")
+
+rpm_config_base_dir="/etc/go-fdo-server"  # RPMs install the default configs here
```

Comment:
This is only used in the cleanup function to remove the rpm default configuration  and this is the only reason why the test is failing.

Please remove this variable and the 382,383 lines and you have it :)

As a side note: the tmt tests are not run in parallel but in serial (alphabetic order I believe) and the first one is removing the default configuration. IIRC there is a way to make all the tests to run in parallel, independently from each other. We can explore that path but for now we need to generate all the configuration before running the test and clean it up alongside the service state (database) at the end, no matter if the test failed or was successful. If we don't do that other tests can fail for unknown reasons.